scipy/integrate/tests/test_cubature.py

```diff
@@ -0,0 +1,927 @@
+import math
+import scipy
+import itertools
+
+import pytest
+import numpy as np
+
+from numpy.testing import assert_allclose
+
+from scipy.integrate._rules import (
+    Rule, FixedRule, ProductNestedFixed,
+    NestedFixedRule, NestedRule,
+    NewtonCotesQuadrature, GaussLegendreQuadrature, GaussKronrodQuadrature,
+    GenzMalikCubature,
+)
+
+from scipy.integrate import cubature
+
```

Comment:
The tests file is slightly difficult to read as it is unclear what is testing the public cubature function and what is testing the rules' functionality. Maybe you organise the tests into classes to make it easier to follow